cmd/wc/wc.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+)
+
+func parseFlagsAndArgument() (string, bool, bool, bool, error) {
+	var lFlag, wFlag, cFlag bool
+
+	flag.BoolVar(&lFlag, "l", false, "count lines flag")
+	flag.BoolVar(&wFlag, "w", false, "count words flag")
+	flag.BoolVar(&cFlag, "c", false, "count characters flag")
+
+	flag.Parse()
+
+	filePath := flag.Arg(0)
+
+	if filePath == "" {
+		return "", false, false, false, fmt.Errorf("invalid file path, you have to specify a valid path to text file")
+	}
+
+	return filePath, lFlag, wFlag, cFlag, nil
+}
+
+func countLines(scanner *bufio.Scanner) {
+	var counter uint = 0
+
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
```

Comment:
this would cause your program to scan a file more than once for most of the time, why not counting all anyways every time and print only the requested info? it